harmony/harmony.py

```diff
@@ -1,4 +1,227 @@
+from harmony.config import Config, Environment
+
+
+class Collection:
+    """The identity of a CMR Collection."""
+
+    def __init__(self, id: str):
+        """Constructs a Collection instance from a CMR Collection ID.
+
+        Parameters
+        ----------
+        id: CMR Collection ID
+
+        Returns
+        -------
+        A Collection instance
+        """
+        self.id = id
+
+
+class BBox(NamedTuple):
+    """A bounding box specified by western & eastern longitude,
+    southern & northern latitude constraints in degrees.
+
+    Example:
+      An area bounded by latitudes 30N and 60N and longitudes
+      130W and 100W:
+
+          >>> spatial = BBox(-130, 30, -100, 60)
+
+      Important: When specified positionally, the parameters must
+      be given in order: west, south, east, north.
+
+      Alternatively, one can explicitly set each bound using the
+      single-letter for each bound:
+
+          >>> spatial = BBox(n=60, s=30, e=-100, w=-130)
+
+      Print the spatial bounds:
+
+          >>> print(spatial)
+          BBox: West:-130, South:30, East:-100, North:60
+
+    Parameters:
+    -----------
+    w: The western longitude bounds (degrees)
+    s: The souther latitude bounds (degrees)
+    e: The easter longitude bounds (degrees)
+    n: The northern latitude bounds (degrees)
+
+    Returns:
+    A BBox instance with the provided bounds.
+    """
+    w: float
+    s: float
+    e: float
+    n: float
+
+    def __repr__(self) -> str:
+        return f'BBox: West:{self.w}, South:{self.s}, East:{self.e}, North:{self.n}'
+
+
+class Request:
+    """A Harmony request with the CMR collection and various parameters expressing how the data is
+    to be transformed.
+
+    Parameters:
+    -----------
+    collection: The CMR collection that should be queried
+    spatial: Bounding box spatial constraints on the data
+    temporal: Date/time constraints on the data
+
+    Returns:
+    --------
+    A Harmony Request instance
+    """
+    def __init__(self, collection: Collection, spatial: BBox = None, temporal: dict = None):
+        self.collection = collection
+        self.spatial = spatial
+        self.temporal = temporal
+        # NOTE: The format is temporary until HARMONY-708:
+        #       https://bugs.earthdata.nasa.gov/browse/HARMONY-708
+        self.format: str = 'image/tiff'
+        self.spatial_validations = [
+            (lambda bb: bb.s < bb.n, 'Southern latitude must be less than Northern latitude'),
+            (lambda bb: bb.s >= -90.0, 'Southern latitude must be greater than -90.0'),
+            (lambda bb: bb.n >= -90.0, 'Northern latitude must be greater than -90.0'),
+            (lambda bb: bb.s <= 90.0, 'Southern latitude must be less than 90.0'),
+            (lambda bb: bb.n <= 90.0, 'Northern latitude must be less than 90.0'),
+            (lambda bb: bb.w < bb.e, 'Western longitude must be less than Eastern longitude'),
```

Comment:
West < East should be accepted as valid, indicating that the bounding box crosses the anti-meridian.

Reply:
Done